pages/first-principles.markdown

```diff
@@ -106,6 +116,8 @@ On the third project, the cycle continues. You're constantly falling behind.
 
 The best option, in my experience, is for the organization to just accept deadlines don't work. That doesn't mean projects should take forever; practice Agile or MVP or what not.
 
+:speech_balloon: Hmm...it seems like the benefits of this principle come from (1) trusting that engineers will choose to make long term investments in the codebase and (2) providing enough autonomy that they can make those choices. If enough space is set aside for this, I don't see a problem with deadlines. In particular, Agile is all about making sprint commitments and hitting your definition of done by the end of the sprint. If those count as deadlines, we had plenty of them at the company where deadlines were rare. (Aside: can we not mention that company by name?) I will agree, however, that deadlines must originate from the people closest to the implementation, not be imposed on them.
```

Comment:
2 weeks?  Wow, that's definitely much shorter than I would have guessed.  I wonder how much of the need for longer term estimates come from the reporting requirements of being a public.  It might be interesting learning more about when and why long term estimates are important.  I'm sure they are in some sets of circumstances (eg, you're writing software that's going on a space probe launching inside of a specific time window), but maybe it's a pretty narrow set of domain-specific reasons.